packages/solana-swap-find-provider/lib/SolanaSwapFindProvider.ts

```diff
@@ -0,0 +1,147 @@
+import { BigNumber, SwapParams, SwapProvider, Transaction } from '@liquality/types'
+import { Provider } from '@liquality/provider'
+import { PendingTxError } from '@liquality/errors'
+import { sha256 } from '@liquality/crypto'
+
+import _filter from 'lodash/filter'
+
+export default class SolanaSwapFindProvider extends Provider implements Partial<SwapProvider> {
+  private instructions = {
+    init: 0,
+    claim: 1,
+    refund: 2
+  }
+
+  async findInitiateSwapTransaction(swapParams: SwapParams): Promise<Transaction<any>> {
+    const { refundAddress } = swapParams
+
+    return await this._findTransactionByAddress({
+      address: refundAddress as string,
+      swapParams,
+      instruction: this.instructions.init,
+      validation: this._compareParams
+    })
+  }
+
+  async findClaimSwapTransaction(swapParams: SwapParams, initiationTxHash: string): Promise<Transaction<any>> {
+    const [initTransaction] = await this.getMethod('getParsedAndConfirmedTransactions')([initiationTxHash])
+
+    if (!initTransaction) {
+      throw new PendingTxError(`Transaction receipt is not available: ${initiationTxHash}`)
+    }
+
+    const {
+      _raw: { buyer }
+    } = initTransaction
+
+    return await this._findTransactionByAddress({
+      swapParams,
+      address: buyer,
+      instruction: this.instructions.claim,
+      validation: this._validateSecret
+    })
+  }
+
+  async findRefundSwapTransaction(swapParams: SwapParams, initiationTxHash: string): Promise<Transaction<any>> {
+    const [initTransaction] = await this.getMethod('getParsedAndConfirmedTransactions')([initiationTxHash])
+
+    if (!initTransaction) {
+      throw new PendingTxError(`Transaction receipt is not available: ${initiationTxHash}`)
+    }
+
+    const {
+      _raw: { seller }
+    } = initTransaction
+
+    return await this._findTransactionByAddress({
+      swapParams,
+      address: seller,
+      instruction: this.instructions.refund
+    })
+  }
+
+  async findFundSwapTransaction(): Promise<null> {
+    return null
+  }
+
+  _compareParams(
```

Comment:
You should use similar method to this one:

https://github.com/liquality/chainabstractionlayer/blob/dev/packages/near-swap-provider/lib/NearSwapProvider.ts#L110